kolibri/plugins/learn/assets/src/state/actions/recommended.js

```diff
@@ -0,0 +1,272 @@
+          store.dispatch('CORE_SET_ERROR', null);
+          store.dispatch('SET_PAGE_NAME', PageNames.LEARN_CHANNEL);
+
+          store.dispatch('CORE_SET_TITLE', `Learn - ${currentChannel.title}`);
+        },
+        error => {
+          handleApiError(store, error);
+        }
+      );
+    },
+    error => {
+      handleApiError(store, error);
+    }
+  );
+}
+
+function showPopularPage(store, channelId) {
+  const state = store.state;
+
+  const pagePrep = Promise.all([
+    _getPopular(channelId, state),
+    setChannelInfo(store, channelId),
+  ]).then(([popular]) => popular, error => error);
+
+  pagePrep.then(
+    popularContent => {
+      const popularPageState = {
+        recommendations: uniqBy(popularContent, 'content_id').map(contentState),
+      };
+      store.dispatch('SET_PAGE_STATE', popularPageState);
+      store.dispatch('SET_PAGE_NAME', PageNames.RECOMMENDED_POPULAR);
+      store.dispatch('CORE_SET_PAGE_LOADING', false);
+      store.dispatch('CORE_SET_ERROR', null);
+
+      const currentChannel = getCurrentChannelObject(state);
+      store.dispatch('CORE_SET_TITLE', `Popular - ${currentChannel.title}`);
+    },
+    error => handleApiError(error)
+  );
+}
+
+function showResumePage(store, channelId) {
+  const state = store.state;
+
+  const pagePrep = Promise.all([
+    _getResume(channelId, state),
+    setChannelInfo(store, channelId),
+  ]).then(([resume]) => uniqBy(resume, 'content_id').map(contentState), error => error);
+
+  pagePrep.then(
+    resumeContent => {
+      const resumePageState = {
+        recommendations: resumeContent,
+      };
+      store.dispatch('SET_PAGE_STATE', resumePageState);
+      store.dispatch('SET_PAGE_NAME', PageNames.RECOMMENDED_RESUME);
+      store.dispatch('CORE_SET_PAGE_LOADING', false);
+      store.dispatch('CORE_SET_ERROR', null);
+
+      const currentChannel = getCurrentChannelObject(state);
+      store.dispatch('CORE_SET_TITLE', `Resume - ${currentChannel.title}`);
+    },
+    error => handleApiError(error)
+  );
+}
+
+function showNextStepsPage(store, channelId) {
+  const state = store.state;
+
+  const pagePrep = Promise.all([
+    _getNextSteps(channelId, state),
+    setChannelInfo(store, channelId),
+  ]).then(([nextSteps]) => uniqBy(nextSteps, 'content_id').map(contentState), error => error);
+
+  pagePrep.then(
+    nextStepsContent => {
+      const nextStepsPageState = {
+        recommendations: nextStepsContent,
+      };
+      store.dispatch('SET_PAGE_STATE', nextStepsPageState);
+      store.dispatch('SET_PAGE_NAME', PageNames.RECOMMENDED_NEXT_STEPS);
+      store.dispatch('CORE_SET_PAGE_LOADING', false);
+      store.dispatch('CORE_SET_ERROR', null);
+
+      const currentChannel = getCurrentChannelObject(state);
+      store.dispatch('CORE_SET_TITLE', `Next Steps - ${currentChannel.title}`);
+    },
+    error => handleApiError(error)
+  );
+}
+
+function showOverviewPage(store, channelId) {
```

Comment:
I think at the very least consolidating the subpages would be worth it, main parameter change seems to be the name of pageState key, and the function for doing the fetch on the content.

It's also possible we could reuse the code in the overall learn page, would require having separate mutations for each key of the pageState, but that would not be the worst thing in the world.

However, for the sake of simplicity, I would just go with the former for now.